src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.cs

```diff
@@ -1015,23 +1001,20 @@ internal unsafe int LastIndexOf(ReadOnlySpan<char> source, ReadOnlySpan<char> va
         /// </summary>
         internal unsafe int IndexOf(string source, string value, int startIndex, int count, CompareOptions options, int* matchLengthPtr, bool fromBeginning = true)
         {
-            Debug.Assert(source != null);
+            StringDebug.AssertInBounds(source, startIndex, count);
             Debug.Assert(value != null);
-            Debug.Assert(startIndex >= 0);
 
             if (matchLengthPtr != null)
             {
                 *matchLengthPtr = 0;
             }
 
+            // All strings trivially begin or end with the empty string.
+            // We'll handle empty-equivalent (zero collation weight) strings later.
+
             if (value.Length == 0)
             {
-                return startIndex;
-            }
-
-            if (startIndex >= source.Length)
-            {
-                return -1;
+                return (fromBeginning) ? startIndex : startIndex + count;
```

Comment:
>count [](start = 67, length = 5)

does the count already validated at this stage to use in this return?

Reply:
never mind I see you already asserting for that.

---
In reply to: [364878392](https://github.com/dotnet/runtime/pull/1514#discussion_r364878392) [](ancestors = 364878392)